src/shared/AssetIcons.tsx

```diff
@@ -18,52 +18,47 @@ import DefaultIcon from '../assets/coins/placeholder.png';
 // The public key of Mykobo's EURC issuer account
 const MYKOBO_ISSUER = 'GAQRF3UGHBT6JYQZ7YSUYCIYWAF4T2SAA5237Q5LIQYJOHHFAWDXZ7NM';
 
-const getStellarAssetIcon = (assetCode: string, assetIssuer: string) => {
-  if (assetCode.includes('AUDD')) {
-    return AUDD;
-  } else if (assetCode.includes('BRL')) {
-    return BRL;
-  } else if (assetCode.includes('EURC')) {
-    if (assetIssuer === MYKOBO_ISSUER) {
-      return mEURC;
-    } else {
-      // Use the Circle EURC icon for all other EURC issuers
-      return cEURC;
-    }
-  } else if (assetCode.includes('NGNC')) {
-    return NGNC;
-  } else if (assetCode.includes('TZS')) {
-    return TZS;
-  } else if (assetCode.includes('USDC')) {
-    return USDC;
-  } else if (assetCode.includes('XLM')) {
-    return XLM;
+const stellarAssets = [
+  { code: 'AUDD', icon: AUDD },
+  { code: 'BRL', icon: BRL },
+  { code: 'NGNC', icon: NGNC },
+  { code: 'TZS', icon: TZS },
+  { code: 'USDC', icon: USDC },
+  { code: 'XLM', icon: XLM },
+];
+
+const polkadotAssets = [
+  { code: 'PEN', icon: PEN },
+  { code: 'DOT', icon: DOT },
+  { code: 'AMPE', icon: AMPE },
+  { code: 'KSM', icon: KSM },
+  { code: 'USDT', icon: USDT },
+  { code: 'GLMR', icon: GLMR },
+];
+
+const getAssetIcon = (assetCode: string, assetIssuer: string, assets: { code: string; icon: string }[]) => {
+  if (assetCode.includes('EURC')) {
+    return assetIssuer === MYKOBO_ISSUER ? mEURC : cEURC;
```

Comment:
I think it can be a bit confusing for a reader, that the definition of `stellarAssets` above doesn't contain EURC.

How about we at least add a comment above `stellarAssets` in l.20 
```
// Maps all supported Stellar assets to icons. The EURC tokens are handled separately in `getAssetIcon()`
```
or similar?

Reply:
This was the reason why I included everything in one function, though the very long `if-else` branches are very ugly, that's for sure. So thanks for this improvement 🙏

Reply:
Good idea @ebma Let's do it

Reply:
What do you think about this? @ebma 

```
// Maps all supported Stellar assets to icons. The EURC tokens are handled separately in `getAssetIcon()`
const stellarAssets = [
  { code: 'AUDD', icon: AUDD },
  { code: 'BRL', icon: BRL },
  { code: 'NGNC', icon: NGNC },
  { code: 'TZS', icon: TZS },
  { code: 'USDC', icon: USDC },
  { code: 'XLM', icon: XLM },
];


const handleSpecialAsset = (assetCode: string, assetIssuer: string) => {
  // The EURC tokens are handled separately because they can be issued by multiple issuers
  if (assetCode.includes('EURC')) {
    return assetIssuer === MYKOBO_ISSUER ? mEURC : cEURC;
  }
};

const getAssetIcon = (assetCode: string, assetIssuer: string, assets: { code: string; icon: string }[]) => {
  const specialAsset = handleSpecialAsset(assetCode, assetIssuer)
  if(specialAsset) return specialAsset;

  const asset = assets.find((asset) => assetCode.includes(asset.code));

  return asset?.icon || undefined;
};
```

Reply:
I love it, let's do it that way 👍